LeanPoly/Poly.lean

```diff
@@ -562,15 +563,120 @@ def tensor.unit.r.bwd {P : Poly} : P ⟶ P ⊗ y :=
   , onDir := λ _ (pdir , _) => pdir
   }
 
+def tensor.leftUnitor (p : Poly) : (y ⊗ p) ≅ p :=
+  { hom := tensor.unit.l.fwd
+  , inv := tensor.unit.l.bwd
+  }
+
+def tensor.rightUnitor (p : Poly) : (p ⊗ y) ≅ p :=
+  { hom := tensor.unit.r.fwd
+  , inv := tensor.unit.r.bwd
+  }
+
+def tensor.associator (p q r : Poly) : (p ⊗ q) ⊗ r ≅ p ⊗ (q ⊗ r) :=
+  { hom := tensor.assoc.bwd
+  , inv := tensor.assoc.fwd
+  }
+
+instance Poly.tensor.monoidalStruct : MonoidalCategoryStruct Poly where
+  tensorObj    := tensor
+  whiskerLeft  := tensor.whiskerLeft
+  whiskerRight := tensor.whiskerRight
+  tensorUnit   := y
+  leftUnitor   := tensor.leftUnitor
+  rightUnitor  := tensor.rightUnitor
+  associator   := tensor.associator
+
+/-- All hypotheses proven automatically so none provided. -/
+instance Poly.tensor.monoidal : MonoidalCategory Poly where
```

Comment:
This is needed for proving closure, since that relies on building a functor out of the tensor product that this instance provides